### PR TITLE
Add tt-ee 2.5.1 authentication features

### PR DESCRIPTION
**Pull request by p7nov**
Resolves https://github.com/tarantool/enterprise_doc/issues/283, https://github.com/tarantool/enterprise_doc/issues/284, https://github.com/tarantool/enterprise_doc/issues/285

- [tt export](https://docs.d.tarantool.io/en/doc/gh-ee283-tt-ee-2-5-1/tooling/tt_cli/export/):
  - add `--token` option and mention it in the subsection about TDG2
  - add `--ssl*`, `--auth`, and `--use-ssl` options
  - add Authentication and Encrypted connection subsections
- [tt import](https://docs.d.tarantool.io/en/doc/gh-ee283-tt-ee-2-5-1/tooling/tt_cli/import/):
  - add `--token` option and mention it in the subsection about TDG2
  - add `--ssl*`, `--auth`, and `--use-ssl` options
  - add Authentication and Encrypted connection subsections

Deployment: https://docs.d.tarantool.io/en/doc/gh-ee283-tt-ee-2-5-1/tooling/tt_cli/export/

### CODE REVIEW

File: doc/tooling/tt_cli/export.rst [hunk 1]
```diff
@@ -168,11 +176,55 @@ Set the ``tt tdg2 export`` batch size less or equal to 1000:
...
+Authentication
+--------------
+
+When connecting to the cluster with enabled authentication, specify access credentials
+in the ``--username`` and ``--password`` command options:
+
+.. code-block:: console
+
+    $ tt crud export localhost:3301 customers:customers.csv \
+                     --username myuser -password p4$$w0rD
```
Comment (DerekBum) [inline doc/tooling/tt_cli/export.rst]:
```suggestion
                     --username myuser --password p4$$w0rD
```

File: doc/tooling/tt_cli/export.rst [hunk 2]
```diff
@@ -168,11 +176,55 @@ Set the ``tt tdg2 export`` batch size less or equal to 1000:
...
+--------------------
+
+To connect to instances that use :ref:`SSL encryption <configuration_connections_ssl>`,
+provide the SSL certificate and SSL key files in the ``--sslcertfile`` and ``--sslkeyfile`` options.
+If necessary, add other SSL parameters in the ``--ssl*`` options.
+
+.. code-block:: console
+
+    $ tt crud export localhost:3301 customers:customers.csv \
+                     --username myuser -password p4$$w0rD   \
```
Comment (DerekBum) [inline doc/tooling/tt_cli/export.rst]:
```suggestion
                     --username myuser --password p4$$w0rD   \
```

File: doc/tooling/tt_cli/export.rst [hunk 3]
```diff
@@ -168,11 +176,55 @@ Set the ``tt tdg2 export`` batch size less or equal to 1000:
...
+                     --auth pap-sha256 --sslcertfile certs/server.crt \
+                     --sslkeyfile certs/server.key
+
+For connections that use SSL but don't require additional parameters, add the ``--use-ssl``
+option:
+
+.. code-block:: console
+
+    $ tt crud export localhost:3301 customers:customers.csv \
+                     --username myuser -password p4$$w0rD   \
```
Comment (DerekBum) [inline doc/tooling/tt_cli/export.rst]:
```suggestion
                     --username myuser --password p4$$w0rD   \
```

File: doc/tooling/tt_cli/export.rst [hunk 4]
```diff
@@ -168,11 +176,55 @@ Set the ``tt tdg2 export`` batch size less or equal to 1000:
...
 .. _tt-export-options:
 
 Options
 -------
 
+..  option:: --auth STRING
+
+    **Applicable to:** ``tt crud export``, ``tt tdg2 export``
+
+    Authentication type: ``chap-sha1`` or ``pap-sha256``.
```
Comment (DerekBum) [inline doc/tooling/tt_cli/export.rst]: There could also be `"auto"` type:
```
	// AutoAuth does not force any authentication method. A method will be
	// selected automatically (a value from IPROTO_ID response or
	// ChapSha1Auth).
```
https://github.com/tarantool/go-tarantool/blob/f9eb9518e070833c7bcb94b8a84c37f7c8af7f15/auth.go#L17-L30

File: doc/tooling/tt_cli/import.rst
Comment (DerekBum) [inline doc/tooling/tt_cli/import.rst:1]: Same comments here:
1. Missing dashes for `password` option.
2. `"auto"` auth type.

File: doc/tooling/tt_cli/export.rst [hunk 5]
```diff
@@ -168,11 +176,55 @@ Set the ``tt tdg2 export`` batch size less or equal to 1000:
...
+.. code-block:: console
+
+    $ tt crud export localhost:3301 customers:customers.csv \
+                     --username myuser -password p4$$w0rD   \
+                     --use-ssl
+
 .. _tt-export-options:
 
 Options
 -------
```
Comment (DerekBum) [inline doc/tooling/tt_cli/export.rst:221]: Don't we also need to write about `--use-ssl` flag somewhere?

**Review by DerekBum — COMMENTED** [review]
Thanks for the doc! Just a few comments, please see below:

File: doc/tooling/tt_cli/export.rst (hunk 4 above)
Comment (p7nov) [reply doc/tooling/tt_cli/export.rst]: Added possible value `auto`

File: doc/tooling/tt_cli/import.rst
Comment (p7nov) [reply doc/tooling/tt_cli/import.rst:1]: Fixed, thanks

File: doc/tooling/tt_cli/export.rst (hunk 5 above)
Comment (p7nov) [reply doc/tooling/tt_cli/export.rst:221]: Added to the options list, linked to the **Encrypted connection** subsection

**Review by DerekBum — APPROVED** [review]

File: doc/tooling/tt_cli/import.rst [hunk 6]
```diff
@@ -377,6 +431,65 @@ Options
     For CSV, double quotes are used by default (``"``).
     The double symbol of this option acts as the escaping symbol within input data.
 
+..  option:: --rollback-on-error
+
+    **Applicable to:** ``tt crud import``
+
+    Specify whether any operation failed on a storage leads to rollback of a batch
+    import on this storage.
```
Comment (lenkis) [inline doc/tooling/tt_cli/import.rst]:
```suggestion
    Specify whether any operation failed on a storage leads to rolling back batch
    import on this storage.
```

Comment (lenkis) [reply doc/tooling/tt_cli/import.rst]: I feel confused with a zero article before "rollback of", as well as with an indefinite article before the abstract noun  "batch import". I'd suggest a rephrasal here.

File: doc/tooling/tt_cli/import.rst [hunk 7]
```diff
@@ -377,6 +431,65 @@ Options
...
+
+    The password for the SSL key file for encrypted connections.
+
+    See also :ref:`tt-import-ssl`.
+
+..  option:: --sslpasswordfile STRING
+
+    **Applicable to:** ``tt crud import``, ``tt tdg2 import``
+
+    A file with list of passwords to the SSL key file for encrypted connections.
```
Comment (lenkis) [inline doc/tooling/tt_cli/import.rst]:
```suggestion
    A file with a list of passwords to the SSL key file for encrypted connections.
```

**Review by lenkis — CHANGES_REQUESTED** [review]

**Review by lenkis — APPROVED** [review]

**Pull request merged**